### PR TITLE
Add issue template for code fix

### PR DESCRIPTION
**Pull request by knshnb**
## Motivation

Currently, we don't have an appropriate issue template for a code fix. This PR adds a code-fix issue template.
Preview:
image
image

## Description of the changes

I added `code-fix.yml`.

### CODE REVIEW

**Comment by c-bata** [comment]
@keisuke-umezawa Could you review this PR?

File: .github/ISSUE_TEMPLATE/code-fix.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    attributes:
+      value: |
+        Thanks for taking the time to raise a new issue! Please write a clear and concise description of the code fix.
+  - type: textarea
+    id: motivation
+    attributes:
+      label: Motivation
+      description: |
+        Please write the motivation for the proposal.
+        If your feature request is related to a problem, please describe a clear and concise description of what the problem is.
```
Comment (nzw0301) [inline .github/ISSUE_TEMPLATE/code-fix.yml:15]: Really minor:

```suggestion
        If your code fix is related to a problem, please describe a clear and concise description of what the problem is.
```

Comment (knshnb) [reply .github/ISSUE_TEMPLATE/code-fix.yml]: Thanks!

**Review by nzw0301 — APPROVED** [review]
Looks great to me. I suppose this issue template makes the Optuna issues well organised!

**Comment by keisuke-umezawa** [comment]
@knshnb 
Although I am not a native English speaker, I doubt that the `Code fix` is a common wording to refer to these kinds of issues. If we use it to refer to the situation that `Restructuring existing computer code without changing its external behavior`, we can use the word `Refactoring` or `Code refactoring`. How about changing the title to `Refactoring` or `Code refactoring`?

**Comment by knshnb** [comment]
Thanks for the comment! That makes sense to me, but I made the name consistent with the existing label ["code-fix"](https://github.com/optuna/optuna/labels/code-fix) (
"Change that does not change the behavior, such as code refactoring."). Do you suggest to change the existing label name as well?

**Comment by knshnb** [comment]
(Sorry, I just deleted a commit that I made by mistake. I didn't make any change by the above force-push.)

**Comment by keisuke-umezawa** [comment]
@knshnb 
In that case, how about changing the description of this issue template as `Suggest a code fix that does not change the behavior, such as code refactoring.? As we already use the existing label name for a long time, we do not need to change it.

**Comment by knshnb** [comment]
Thanks! I applied your comment.

**Review by keisuke-umezawa — APPROVED** [review]
LGTM! Thank you for working on it.

**Comment by keisuke-umezawa** [comment]
@knshnb Could you rebase against master branch? If all tests pass, you can merge it by yourself.

**Comment by knshnb** [comment]
Let me merge this PR as @keisuke-umezawa suggested!

**Pull request merged**